chapters/functions.tex

```diff
@@ -284,6 +284,12 @@ \section{Pure Modelica Functions}\label{pure-modelica-functions}
 The reason is that solving systems of equations generally requires expressions to be evaluated an unknown number of times.
 This includes the special handling of \lstinline!when initial()! during initialization.
 
+There are two ways in which an impure function could be called in a system of equations,
+namely in the deprecated case of external functions assumed to be impure, and when using \lstinline!pure($\ldots$)! to call an impure function from within a pure function.
+The side-effect semantics are undefined in this case.
```

Comment:
Just asking: Is it clear that _this case_ refers to both ways in the impure function could be called?

Reply:
OK, let me turn that into a concrete suggestion:
```suggestion
The side-effect semantics are undefined in these cases.
```
or avoid _this case_ by writing:
```suggestion
The side-effect semantics of the function call are then undefined.
```